frontend/apps/marketing/src/components/contentful/activityCatalog/FilterBar/FacetDropdown.tsx

```diff
@@ -0,0 +1,196 @@
+
+'use client';
+import CloseRoundedIcon from '@mui/icons-material/CloseRounded';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import { Box, Button, Chip, Collapse, Menu } from '@mui/material';
+import ClickAwayListener from '@mui/material/ClickAwayListener';
+import { clsx } from 'clsx';
+import * as React from 'react';
+
+import styles from './facetDropdown.module.scss';
+
+type Props = {
+  facetKey: string;
+  label: string;
+  options: string[];
+  selected: Set<string> | undefined;
+  onChange: (facetKey: string, next: Set<string>) => void;
+  onOpenChange?: (open: boolean) => void;
+  variant?: 'menu' | 'inline';
+};
+
+export default function FacetDropdown({
+  facetKey,
+  label,
+  options,
+  selected,
+  onChange,
+  onOpenChange,
+  variant = 'menu',
+}: Props) {
+  // ----- hooks (top) -----
+  const sel = selected ?? new Set<string>();
+  const isActive = sel.size > 0;
+
+  // Inline
+  const [openInline, setOpenInline] = React.useState(false);
+
+  // Menu
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const [anchorWidth, setAnchorWidth] = React.useState<number | undefined>();
+  const menuOpen = Boolean(anchorEl);
+
+  React.useEffect(() => {
+    if (!menuOpen || !anchorEl) return;
+    const update = () => setAnchorWidth(anchorEl.offsetWidth);
+    update();
+    window.addEventListener('resize', update);
+    return () => window.removeEventListener('resize', update);
```

Comment:
Rather than use a window resize listener, can the dropdown be natively responsive to avoid the usage of JS to handle this?